### PR TITLE
Replace function save to join/exit

### PR DESCRIPTION
**Pull request by gbalabasquer**
A couple of doubts:
- I've named `subi` to the function that subtracts from two `uint` and returns an `int`. @rainbreak probably will not like it :). Any better ideas to solve this name collision?
- What is the idea behind the `move` function in the `Pot`? Are we planning to use for any specific case? or just in case?

### CODE REVIEW

**Comment by rainbreak** [comment]
@gbalabasquer `move` was for making Savings Dai transferable. I'd say delete it, unless there's a requirement?

**Comment by gbalabasquer** [comment]
IMO we can get rid of it or implement a `move` with `wish` or a traditional `transferFrom` approach. The actual `move` function doesn't work per se, it will need some authorized wrapper to handle the transfers and I don't think it is necessary to make it more complicated. Excepting we want to leave it in the core of `dsr` for a future usage but avoiding it can be used in the beginning.

**Comment by rainbreak** [comment]
Ok, then I say delete it. That should let you remove some redundant math functions. As for `subi`, you could call it `diff` like in the jug.

**Comment by gbalabasquer** [comment]
Just made the changes. Let me know what you think and if we could merge.
If we need the transfer function we can evaluate later the best way to implement it.

**Comment by gbalabasquer** [comment]
There is one more open question I had, that is about `drip`.
I think we should include it in `join` and `exit` but I I recall correctly there was some FV complication. Is it really the case or can we add it?

**Comment by rainbreak** [comment]
For now `drip` needs to remain separate, but we'd expect resolution on this this month.

**Comment by rainbreak** [comment]
@gbalabasquer yes, do you want to update to `suck` separately?

**Comment by gbalabasquer** [comment]
@rainbreak ahh ok sorry I hadn't reviewed the changes in `dss` yet. Will look into that so `dsr` uses the latest dependency version.

**Comment by gbalabasquer** [comment]
@rainbreak `suck` is now there.

File: src/dsr.sol [hunk 1]
```diff
@@ -125,14 +113,12 @@ contract Pot is DSNote {
...
-    function diff(uint x, uint y) internal pure returns (int z) {
-        z = int(x) - int(y);
-        require(int(x) >= 0 && int(y) >= 0);
-    }
-
     // --- Administration ---
     function file(bytes32 what, uint256 data) public note auth {
-        if (what == "dsr") dsr = data;
+        if (what == "dsr") {
+            require(data >= ONE);
```
Comment (rainbreak) [inline src/dsr.sol]: I think this is redundant now that you use `sub(uint,uint) returns (uint)` in `drip`.

Comment (gbalabasquer) [reply src/dsr.sol]: Yeah, in the meanwhile `drip` remains outside of `join` and `exit` the `sub` will just do the trick. However if in the future we end up calling it those functions, setting a wrong `dsr`s will make the functions to fail. Probably not a big deal, so yeah, I'll remove it.

Comment (gbalabasquer) [reply src/dsr.sol]: Done.

**Pull request merged**